server/include/bge/System.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "Entity.h"
+#include "ComponentManager.h"
+#include "Component.h"
+#include <set>
+#include <bitset>
+
+namespace bge {
+    class System {
+        public:
+            void init();
+            void update(int dt);
+        
+            void registerEntity(Entity entity);
+            void deRegisterEntity(Entity entity);
+
+        protected:
+            std::bitset<32> systemSignature;
```

Comment:
What is systemSignature for?

Reply:
https://austinmorlan.com/posts/entity_component_system/#the-signature

Although we might not need this since we already have a vector of registeredEntities